pkg/services/ngalert/ngalert.go

```diff
@@ -417,7 +417,10 @@ func (ng *AlertNG) init() error {
 	}
 	logger := log.New("ngalert.state.manager.persist")
 	statePersister := state.NewSyncStatePersisiter(logger, cfg)
-	if ng.FeatureToggles.IsEnabledGlobally(featuremgmt.FlagAlertingSaveStatePeriodic) {
+
+	if ng.FeatureToggles.IsEnabledGlobally(featuremgmt.FlagAlertingSaveStateCompressed) {
+		statePersister = state.NewSyncRuleStatePersisiter(logger, cfg)
```

Comment:
`SyncRuleStatePersister` is similar to `SyncStatePersister` but operates at the rule level. Instead of saving each alert instance individually using `SaveAlertInstance`, it saves the entire rule state at once using `SaveAlertInstancesForRule`.